### ISSUE TITLE
Clarity issues in the mixed-precision execution guide

### ISSUE THREAD

**Issue by Antymon** [tensorflow/tensorflow#50870]
## URL(s) with the issue:

In particular this point: https://github.com/tensorflow/docs/blame/master/site/en/guide/mixed_precision.ipynb#L817

### Clear description

Given linked summary point:

>   If your model ends in softmax, make sure it is float32. And regardless of what your model ends in, make sure the output is float32.

How do exactly those sentences make sense together?

If "output" is a result of an arbitrary calculation then why mention specific softmax at all?
If "output" is a variable then it was mentioned before that all variables should be float32, and pointing at the end specifically doesn't make much sense either.

I find this part of the summary very confusing.

As a side note, if there was any reference to explain why despite float16 calculation it still makes sense to store the
output in float32 variables that would also be very useful. I.e., why it helps with numeric precision exactly?. I mean, low-resolution operation presumably results in a low-resolution result, therefore how is it not wasteful to store such result in a higher resolution "container"? Clearly, it is not wasteful because "everyone" does it.

**Comment by tilakrayal** [comment]
@Antymon ,

Can you please take a look at this [link](https://www.tensorflow.org/guide/mixed_precision) for information on mixed-precision.It helps.Thanks

**Comment by Antymon** [comment]
@tilakrayal Sorry, but I don't follow your suggestion - the issue is precisely on this piece of text, it quotes pieces of it.

**Comment by ChrisChiasson** [comment]
Maybe the reason is that they want the _input_ to the softmax to be float32 because of the loss of precision associated with exponentiation. (I don't work here, I was just looking for someone to answer my own issue because mine isn't being answered.)

**Comment by Saduf2019** [comment]
@Antymon 
The answer in the notebook, as it says calculating loss  with float 16 precision may result in numeric loss.
This is not a bug or feature request, for any further queries you may open this issue in tf discussion forum as there is a larger community there.
